Exercise_1.java

```diff
@@ -1,33 +1,51 @@
+
 class Stack { 
     static final int MAX = 1000; 
     int top; 
     int a[] = new int[MAX]; // Maximum size of Stack 
   
     boolean isEmpty() 
-    { 
-        //Write your code here 
+    {
+		return top == -1;
     } 
 
     Stack() 
     { 
-        //Initialize your constructor 
+       top = -1;
     } 
   
     boolean push(int x) 
-    { 
+    {
         //Check for stack Overflow
         //Write your code here
+    	if(top < MAX)
+    	{
+    		a[++top] = x;
+    		
+    		return true;
+    	}
+    	System.out.println("Stack Overflow");
+    	return false;
     } 
   
     int pop() 
-    { 
+    {
+		
         //If empty return 0 and print " Stack Underflow"
         //Write your code here
+    	if(top >= 0)
+    	{
+    		int x= a[top];
+    		a[top--] = 0;
+    		return x;
+    	}
+    	System.out.println("Stack Underflow");
+    	return 0;
     } 
   
     int peek() 

```

Comment:
Please add edge case to prevent from operating on an empty stack.

Reply:
If the stack is empty, top will be -1. If top is not -1, I am accessing stack elements. (This is the edge case. please let me know if I have to add anything)